File: assign2/erlang/src/tail_recursive.erl
-module(tail_recursive).
-export([tailRecursiveFib/1]).

tailRecursiveFib(0) -> [1];
tailRecursiveFib(1) -> [1, 1];
tailRecursiveFib(Position) when Position > 1 -> tailRecursiveFib(Position - 1, [1, 1]).

tailRecursiveFib(0, Series) -> Series; 
tailRecursiveFib(StepsLeft, Series) ->
  tailRecursiveFib(StepsLeft - 1, Series ++ [lists:nth(length(Series), Series) + lists:nth(length(Series) - 1, Series)]).

File: assign2/ocaml/src/tailRecursive.ml
let tailRecursiveFib position =
  let rec fib_list count series =
    if count <= 0 then series 
    else
      let length = List.length series in
      let cur = List.nth series (length - 1) in
      let prev = List.nth series (length - 2) in

      fib_list (count - 1) (series @ [prev + cur]) 
  in
  match position with
    | 0 -> [1]
    | _ -> fib_list (position - 1) [1; 1]
